diff --git a/omnn/math/Polyfit.h b/omnn/math/Polyfit.h
index 1234567..89abcdef 100644
--- a/omnn/math/Polyfit.h
+++ b/omnn/math/Polyfit.h
@@ -17,6 +17,7 @@

 #include <boost/algorithm/string/replace.hpp>
 #include <boost/lexical_cast.hpp>
+#include <boost/numeric/ublas/storage_wrapper.hpp>
 #include <boost/numeric/ublas/matrix.hpp>
 #include <boost/numeric/ublas/lu.hpp>

@@ -47,8 +48,9 @@ auto polyfit( const std::vector<T>& oX, const std::vector<T>& oY, size_t nDegree )
     nDegree++;

     auto nCount = oX.size();
-    matrix<T> oXMatrix( nCount, nDegree );
-    matrix<T> oYMatrix( nCount, 1 );
+    using wrapped_allocator = allocator_wrapper<std::allocator<T>>;
+    matrix<T, row_major, unbounded_array<T, wrapped_allocator>> oXMatrix(nCount, nDegree);
+    matrix<T, row_major, unbounded_array<T, wrapped_allocator>> oYMatrix(nCount, 1);

     // copy y matrix
     for ( size_t i = 0; i < nCount; i++ )
